tests/src/test/java/me/snowdrop/istio/dt/OpenshiftIT.java

```diff
@@ -1,129 +1,152 @@
+        jaegerTracesURL = "https://" + getRouteInIstioNamespace(JAEGER_QUERY_ROUTE) + "api/traces";
 
-    @After
-    public void cleanup() {
-        if (routeRule != null){
-            istioAssistant.undeployIstioResources(routeRule);
-        }
+        waitUntilApplicationIsReady();
     }
 
+
     /**
      * Interacts with application and test if traces are present in jaeger
-     * @throws ParseException
      * @throws InterruptedException
      */
     @Test
-    public void tracingTest() throws ParseException, InterruptedException {
+    public void tracingTest() throws InterruptedException {
         Long startTime = TimeUnit.MILLISECONDS.toMicros(System.currentTimeMillis());
 
-        RestAssured.get(istioURL + "greeting/api/greeting");
+        RestAssured.get(istioIngressGatewayRouteURL + "greeting/api/greeting");
 
         TimeUnit.SECONDS.sleep(10); // wait 10 seconds so span will show up in the jaeger
 
-        JSONArray jaegerTraces = getJaegerTraces(startTime);
-        Assert.assertEquals("Query should create one trace",1,jaegerTraces.size());
-        int spanCount = ((JSONArray)((JSONObject)jaegerTraces.get(0)).get("spans")).size();
-        Assert.assertTrue("Trace should have at least 4 spans, has: " + spanCount, spanCount >= 4);
-    }
+        RestAssured
+                .given()
+                .relaxedHTTPSValidation() //jaeger uses https, so we need to trust the cert
+                .param("service", ISTIO_INGRESS_GATEWAY_SERVICE_NAME)
+                .param("start",startTime)
+                .get(jaegerTracesURL)
+                .then()
+                .statusCode(200)
+                /*
+                    data[0].processes should contain something like:
+
+                    "p1": {
+                        "serviceName": "istio-ingressgateway",
+                        "tags": [
+                            {
+                                "key": "ip",
+                                "type": "string",
+                                "value": "172.17.0.8"
+                            }
+                        ]
+                    },
+                    "p2": {
+                        "serviceName": "istio-policy",
+                        "tags": [
+                            {
+                                "key": "ip",
+                                "type": "string",
+                                "value": "172.17.0.11"
+                            }
+                        ]
+                    }
+
+                    ...
+
+                 */
+                .body("data[0].processes", new TypeSafeMatcher<Map<String, Object>>() {
+
+                    @Override
+                    public void describeTo(Description description) {}
+
+                    @Override
+                    protected boolean matchesSafely(Map<String, Object> item) {
+                        return ensureProcessesMapContainsBothServices(item);
```

Comment:
Why can't this method define all the checking logic?

Reply:
It could, but I wanted to give it an explicit name because `matchesSafely` is a little to generic. I wanted to convey the message that a check for both services is going to be performed